p2p/net/mock/mock_test.go

```diff
@@ -598,7 +598,9 @@ func TestFuzzManyPeers(t *testing.T) {
 		peerCount = 100
 	}
 	for i := 0; i < peerCount; i++ {
-		_, err := FullMeshConnected(context.Background(), 2)
+		ctx, cancel := context.WithCancel(context.Background())
+		_, err := FullMeshConnected(ctx, 2)
+		cancel()
```

Comment:
the need for this seems to indicate there are still tasks attached to the context by this  process that aren't cleaned up as we might hope. canceling the context solves this problem and is good to be able to do, but we should probably mark the underlying laziness for cleanup as well.

Reply:
`FullMessConnected` returns a fully connected network. Unfortunately, the "correct" way to close that network is to cancel the context. There's no explicit close function.